### PR TITLE
Improve docstring of Axes.hexbin

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As part of #10148: Docstring update for `Axes.hexbin`.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4496,63 +4496,60 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
...
-            If 'log', use a logarithmic scale for the color
-            map. Internally, :math:`log_{10}(i+1)` is used to
-            determine the hexagon color.
+            Alternatively, if a tuple (*nx*, *ny*), the number of hexagons
+            in the *x*-direction and the *y*-direction.
 
-            If an integer, divide the counts in the specified number
-            of bins, and color the hexagons accordingly.
+        bins : 'log' or int or sequence, default: *None*
+            Discretation of the hexagon values.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
            Discretization of the hexagon values.
```

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -4496,63 +4496,60 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
...
 
-        mincnt : int > 0, optional, default is *None*
+        mincnt : int > 0, default: *None*
             If not *None*, only display cells with more than *mincnt*
-            number of points in the cell
+            number of points in the cell.
 
-        marginals : bool, optional, default is *False*
+        marginals : bool, default: *False*
             if marginals is *True*, plot the marginal density as
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
            If marginals is *True*, plot the marginal density as
```

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -4565,54 +4562,67 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
...
-            fill color.
+            - 'face': Draw the edges in the same color as the fill color.
+            - 'none': No edges are drawn. This can sometimes lead to unsightly
+              unpainted pixels between the hexagons.
+            - *None*: Draw outlines in the default color.
+            - An explicit matplotlib color.
 
-            If 'none', no edge is drawn; this can sometimes lead to unsightly
-            unpainted pixels between the hexagons.
+        reduce_C_function : callcable, default is `numpy.mean`
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
        reduce_C_function : callable, default is `numpy.mean`
```

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -4565,54 +4562,67 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
...
+            All other keyword arguments are passed on to `.PolyCollection`:
+
+            %(PolyCollection)s
 
         Returns
         -------
         polycollection
-            A `.PolyCollection` instance; use `.PolyCollection.get_array` on
-            this to get the counts in each hexagon.
+            A `.PolyCollection` defining the hexagonal bins.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Type can be placed on the name line?

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -4565,54 +4562,67 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
...
+
+            - `.Polycollection.get_offest` contains a Mx2 array containing
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
            - `.PolyCollection.get_offset` contains a Mx2 array containing
```

File: lib/matplotlib/axes/_axes.py (hunk 4 above)
Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: Added the type above. Nevertheless, I'd leave this line as is.

**Pull request merged**